src/cudamatrix/cu-math.cc

```diff
@@ -205,6 +206,31 @@ void Copy(const CuMatrixBase<Real> &src, const CuArray<int32> &copy_from_indices
   }
 }
 
+template<typename Real>
+void ComputeXvectorObjfFromScores(const CuMatrixBase<Real> &scores,
+                                  CuMatrixBase<Real> *objf_terms,
+                                  CuMatrixBase<Real> *objf_derivs) {
+  #if HAVE_CUDA == 1
```

Comment:
check the dimensions at the beginning of this function-- KALDI_ASSERT(SameDim(scores, *objf_terms) && ..)